db.js

```diff
@@ -449,21 +449,56 @@ var loans = (function () {
       let updateInserts = [loan.status, loan.id];
       updateQuery = mysql.format(updateQuery, updateInserts);
 
-      //var now = moment().format("YYYY-MM-DD HH:mm:ss");
-      //will be used later for timestamps
-      //console.log(now);
-
-      return pool.query(updateQuery).then((updateResult) => {
-        if (loan.status === "accepted") {
-          let deleteQuery =
-            "DELETE FROM loans WHERE status = 'pending' AND book_id = ?";
-          let deleteInserts = [loan.book_id];
-          deleteQuery = mysql.format(deleteQuery, deleteInserts);
-
-          pool.query(deleteQuery);
-        }
-        return updateResult;
-      });
+      var currentDate = moment().format("YYYY-MM-DD HH:mm:ss");
+
+      var updateTimeQuery = "UPDATE loans SET ";
+      // After createing an update query, fetching the current time, and creating a starting updateTime query,
+      // check to see if the status is anything but "accepted." If so, set the time query to update its respective field.
+
+      if (loan.status === "loaned") {
+        updateTimeQuery = updateTimeQuery + "loan_start_date = ? WHERE id = ?";
+        let updateTimeInserts = [currentDate, loan.id];
+        updateTimeQuery = mysql.format(updateTimeQuery, updateTimeInserts);
+
+        pool.query(updateTimeQuery);
+      } else if (loan.status === "returned") {
+        updateTimeQuery = updateTimeQuery + "return_date = ? WHERE id = ?";
+        let updateTimeInserts = [currentDate, loan.id];
+        updateTimeQuery = mysql.format(updateTimeQuery, updateTimeInserts);
+
+        pool.query(updateTimeQuery);
+      } else if (loan.status === "canceled" || loan.status === "declined") {
+        var queuedForDeletion = 1; // Sets a delete flag
+        var deleteQuery = "DELETE FROM loans WHERE id = ?";
+        var inserts = [loan.id];
+        deleteQuery = mysql.format(deleteQuery, inserts);
+      }
+
+      if (queuedForDeletion === 1) {
```

Comment:
Isn't `queuedForDeletion` out of scope here? Also, it might be easier to use a boolean

Reply:
Thanks, I like the idea of a switch statement much more than a string of ifs. In regards to the `deleteLoan()` function, since it returns `pool.query()`, wouldn't that query be what is returned instead of the `updateQuery`? I initially started out writing this commit using `deleteLoan()`, but that was what seemed to be happening during testing.

Also, as to the scope of `queuedForDeletion`, it is not out of scope because `queuedForDeletion` is declared using`var`, not `let`.